### PR TITLE
feat: wire editor image upload to active image plugin

### PR DESCRIPTION
**Pull request by Svtter**
## Summary
When a running plugin provides `image_upload` capability, editor image uploads now go through the plugin's gRPC `ImageUploader` service instead of saving to local disk. Falls back to local save when no image plugin is available.

## Changes
- `routes/image_routes.py` — Added `_try_plugin_upload()` helper and wired it into the `upload_image()` route
  - Discovers running plugins with `image_upload` capability
  - Reads file bytes, splits into 64 KiB chunks for gRPC client-streaming
  - Returns the plugin-provided URL on success
  - Falls back to existing local save behavior when no plugin available

## How it works
1. Check `registry.plugin_manager` for any running plugin with `image_upload` capability
2. Get gRPC `ImageUploaderStub` for that plugin
3. Stream file in 64 KiB chunks via gRPC `Upload` RPC
4. Return the URL from the plugin response
5. If anything fails or no plugin available, fall back to local `save_image()`

### CODE REVIEW

File: routes/image_routes.py [hunk 1]
```diff
@@ -3,12 +3,79 @@
...
+        return None
+
+    stub = plugin_manager.get_image_uploader_stub(target["name"])
+    if stub is None:
+        return None
+
+    try:
+        from proto import plugin_pb2
+
+        file_bytes = file_storage.read()
```
Comment (Svitter) [inline routes/image_routes.py:46]: **blocking**: `file_storage.read()` slurps the entire file into memory with no size check. The local `save_image` fallback enforces `MAX_IMAGE_SIZE` (20 MiB), but the plugin path has no guard. A malicious or accidental multi-GB upload will exhaust server memory before any gRPC frame is sent.

Consider reading up to `MAX_IMAGE_SIZE + 1` bytes first, or checking `Content-Length` from the request, and short-circuiting oversized files before the read.

File: routes/image_routes.py [hunk 2]
```diff
@@ -3,12 +3,79 @@
...
+
+    if target is None:
+        return None
+
+    stub = plugin_manager.get_image_uploader_stub(target["name"])
+    if stub is None:
+        return None
+
+    try:
+        from proto import plugin_pb2
```
Comment (Svitter) [inline routes/image_routes.py:44]: **should-fix**: This `from proto import plugin_pb2` is a dynamic import inside a `try` block. Project conventions require top-level imports only — no inline `import()` or conditional imports. Move to the top of the file alongside the other imports. The `proto` package is always available (it's part of this repo), so there's no risk of import failure that warrants the guard.

**Review by Svitter — COMMENTED** [review]
P1 — mergeable after a touch. Tight, well-scoped feature with a clean fallback chain. Two findings to address before merge:

**blocking**: `_try_plugin_upload()` at `routes/image_routes.py:46` slurps the entire file into memory with no size check. The local `save_image` path enforces `MAX_IMAGE_SIZE` (20 MiB), but the plugin path doesn't — a large upload can OOM the server.

**should-fix**: Dynamic `from proto import plugin_pb2` at line 44 violates the no-inline-import convention. Move it to the top-level imports.

No test coverage for the new function (5+ branches: no plugin manager, no running plugin, no stub, upload success, upload failure). Worth adding a unit test that exercises the fallback decisions even if you mock at the gRPC boundary.

Thanks for the clean, focused PR — the gRPC streaming chunk pattern looks right, and the graceful fallback to local save is well-handled.
